lib/puppet/util/windows/registry.rb

```diff
@@ -213,7 +213,7 @@ def read(key, name_ptr, *rtype)
         when Win32::Registry::REG_MULTI_SZ
           result = [ type, data_ptr.read_wide_string(string_length).split(/\0/) ]
         when Win32::Registry::REG_BINARY
-          result = [ type, data.read_bytes(0, byte_length) ]
+          result = [ type, data_ptr.read_bytes(0, byte_length) ]
```

Comment:
I was able to get a hold of the data causing the problem.  The call needs to be changed to:

```ruby
result = [ type, data_ptr.read_bytes(byte_length) ]
```

Note the lack of a `0` there.